packages/controls/src/controls/rich-text/v2/plugin.ts

```diff
@@ -0,0 +1,22 @@
+import { type Editor, type Element, type Text } from 'slate'
+
+import { type Data } from '../../../common'
+import { type DataType } from '../../associated-types'
+import { ControlDefinition } from '../../definition'
+
+export type RichTextPluginControl<
+  Def extends ControlDefinition = ControlDefinition,
+  ValueType extends Data = Data,
+> = {
+  definition: Def
+  getValue?(editor: Editor): ValueType | undefined
+  onChange?(editor: Editor, value: ValueType): void
```

Comment:
Made `getValue` and `onChange` optional here so we can serialized/deserialize `RichTextV2` definition w/o having to provide/generate "deserialized plugin" type -- see the corresponding serialization code here: https://github.com/makeswift/makeswift/pull/781/files#diff-43f3bf8883378456d53431e89b27f0c02e470722d8ae9e3ebcba9c6008fe32d4R147